ethdb/privateapi/ethbackend.go

```diff
@@ -505,26 +492,30 @@ func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *r
 	headHash := rawdb.ReadHeadBlockHash(tx2)
 	headNumber := rawdb.ReadHeaderNumber(tx2, headHash)
 	headHeader := rawdb.ReadHeader(tx2, headHash, *headNumber)
-	tx2.Rollback()
 
 	if headHeader.Hash() != forkChoice.HeadBlockHash {
+		tx2.Rollback()
 		return nil, fmt.Errorf("unexpected head hash: %x vs %x", headHeader.Hash(), forkChoice.HeadBlockHash)
 	}
 
 	if headHeader.Time >= req.PayloadAttributes.Timestamp {
+		tx2.Rollback()
```

Comment:
No defer?

Reply:
No, because later in the code `BlockBuilder` is responsible for rolling back `tx2` (which happens in a different goroutine).

Reply:
reverted